### PR TITLE
remove linear search in get_base_map_name and get_base_hero_name + all other data lists.

### PR DESCRIPTION
**Pull request by CamDavidsonPilon**
While not the least-performant part of the codebase, I wanted to clean up `get_base_hero_name` and `get_base_map_name`. They were doing inefficient loops over lists, and these can be much faster by using dict lookups. I've created classes for these lookups too, as they can encapsulate all the translation logic we require. 

Similarly, all the lists in `data/__init__.py` could be sets which offer better `in` performance. 

I've also add a test folder, I can run it locally with:
```
$ pip install pytest
$ py.test tests/
```

I also tested these changes locally in my database, things look good there too.

I'm most concerned about not catching unicodes correctly in production.

### CODE REVIEW

File: data/__init__.py [hunk 1]
```diff
@@ -151,377 +151,403 @@
...
+                                    u"\uac00\uc988\ub85c /// Gazlowe", u"Gazol /// Gazlowe",
+                                    u"\u0413\u0430\u0437\u043b\u043e\u0443", u"\u52a0\u5179\u9c81\u7ef4 /// Gazlowe",
+                                    u"\u52a0\u8332\u9b6f\u7dad /// Gazlowe"],
+                         "Tracer": [u"Tracer", u"Tracer /// Tracer", u"Tracer /// Tracer", u"Tracer",
+                                    u"Tracer /// Tracer", u"Tracer", u"Tracer",
+                                    u"\ud2b8\ub808\uc774\uc11c /// Tracer", u"Smuga",
+                                    u"\u0422\u0440\u0435\u0439\u0441\u0435\u0440", u"\u730e\u7a7a",
+                                    u"\u9583\u5149 /// Tracer"],
+                         "Tychus": [u"Tychus", u"Tychus", u"Tychus /// Tychus", u"Tychus /// Tychus",
+                                    u
```
Comment (CamDavidsonPilon) [inline data/__init__.py]: so I'm not recreating to inverse mapping each time a new HeroUnit or Player class is created.

File: data/__init__.py [hunk 2]
```diff
@@ -151,377 +151,403 @@
...
+                         "Valla": [u"DemonHunter", u"Valla", u"Valla /// Valla", u"Valla /// Valla",
+                                   u"Vala /// Valla", u"Valla /// Valla", u"Valla", u"\ubc1c\ub77c /// Valla",
+                                   u"Valla /// Valla", u"\u0412\u0430\u043b\u043b\u0430",
+                                   u"\u7ef4\u62c9 /// Valla", u"\u7dad\u62c9 /// Valla"],
+                         "Diablo": [u"Diablo", u"Diablo /// Diablo", u"Diablo /// Diablo", u"Diablo /// Diablo",
+                                    u"Diablo /// Diablo", u"Diablo /// Diablo", u"Diablo /// Diablo",
+                                    u"\ub514\uc544\ube14\ub85c /// Diablo", u"Diablo /// Diablo",
+                                    u"\u0414\u0438\u0430\u0431\u043b\u043e /// Diablo",
+                                    u"\u8fea\u4e9a\u6ce2\u7f57 /// Diablo", u"\u8fea\u4e9e\u5e03\u7f85 /// Diablo"],
+                         "Lunara": [u"Dryad", u"Lunara /// Lunara", u"Lunara /// Lunara", u"Lunara /// Lunara",
```
Comment (crorella) [inline data/__init__.py:303]: I don't get why you replaced the key from Dryad to Lunara.

File: tests/data/test_data.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+def test_HeroTranslator_with_english_characters():
+    assert HeroTranslator.get_base_hero_name('Falstad') == 'Falstad'
+    assert HeroTranslator.get_base_hero_name('Zuljin') == "Zul'jin"
+    assert HeroTranslator.get_base_hero_name('Tinker') == "Gazlowe"
+    assert HeroTranslator.get_base_hero_name(u"Gazleu /// Gazlowe") == "Gazlowe"
+    assert HeroTranslator.get_base_hero_name("Gazleu /// Gazlowe") == "Gazlowe"
+    assert HeroTranslator.get_base_hero_name(u"Thrall /// Thrall") == "Thrall"
+    assert HeroTranslator.get_base_hero_name("Thrall /// Thrall") == "Thrall"
+
```
Comment (crorella) [inline tests/data/test_data.py]: Can you please add tests for Gul'dan, Brightwing and Lunara? The reason is the internal representation for these heroes are Guldan, FaerieDragon and Dryad respectively and I would like to test those cases.

File: data/__init__.py (hunk 2 above)
Comment (CamDavidsonPilon) [reply data/__init__.py:303]: The keys are now the "canonical" name of the hero - so Lunara should be placed there. This is the same logic as previous code - if Dryad was the key and "Dryad" as inputted, we still would have outputted Lunara - see the previous code here: https://github.com/crorella/hots-parser/pull/5/files/ee572c1086ccffd6a211b41d2e3579db1608f621#diff-740b116474b5b8b4d5927ca113397b13L18

File: tests/data/test_data.py (hunk 3 above)
Comment (CamDavidsonPilon) [reply tests/data/test_data.py]: Done - tests pass locally

**Comment by crorella** [comment]
Okay, I also tested locally and seems to be working fine, thanks @CamDavidsonPilon !!

**Pull request merged**